src/snowflake/connector/result_batch.py

```diff
@@ -270,18 +278,33 @@ def _download(self, **kwargs) -> "Response":
         for retry in range(MAX_DOWNLOAD_RETRY):
             try:
                 with TimerContextManager() as download_metric:
+                    chunk_url = self._remote_chunk_info.url
                     response = requests.get(
-                        self._remote_chunk_info.url,
+                        chunk_url,
                         headers=self._chunk_headers,
                         timeout=DOWNLOAD_TIMEOUT,
                         stream=True,
                     )
-                    if response.ok:
+
+                    if response.status_code == OK:
                         break
-            except Exception as e:
+
+                    # Raise error here to correctly go in to exception clause
+                    if is_retryable_http_code(response.status_code):
+                        # retryable server exceptions
+                        error: Error = get_http_retryable_error(response.status_code)
+                        raise RetryRequest(error)
+                    elif response.status_code == UNAUTHORIZED:
+                        # make a unauthorized error
+                        raise_okta_unauthorized_error(None, response)
+                    else:
+                        raise_failed_request_error(None, chunk_url, "get", response)
+
+            except (RetryRequest, Exception) as e:
+                e = e.args[0] if isinstance(e, RetryRequest) else e
```

Comment:
This "unwraps" the underlying error so that the error we raise is the true HTTPError that occurred.